implementations/rust/ockam/ockam_api/tests/portals.rs

```diff
@@ -0,0 +1,628 @@
+                    "127.0.0.1:0".to_string(),
+                    "inlet_alias".to_string(),
+                    route![],
+                    route![],
+                    second_node_listen_address
+                        .multi_addr()?
+                        .concat(&MultiAddr::from_str("/secure/api/service/outlet")?)?,
+                    None,
+                    None,
+                    None,
+                    true,
+                )
+                .await?;
+
+            info!("inlet_status: {inlet_status:?}");
+
+            // connect to inlet_status.bind_addr and send dummy payload
+            const PAYLOAD_SIZE: usize = 50 * 1024 * 1024;
+            info!("generating random payload");
+            let payload = {
+                let mut payload = vec![0u8; PAYLOAD_SIZE];
+                rand::thread_rng().fill_bytes(&mut payload);
+                payload
+            };
+
+            let stream = TcpStream::connect(inlet_status.bind_addr.clone())
+                .await
+                .unwrap();
+
+            // saturate the bandwidth with empty packets
+            // and verify the connection stays up for 30 seconds
+            let (mut rx, mut tx) = stream.into_split();
+
+            let payload_cloned = payload.clone();
+            // keeps a reference to the connection to avoid it being dropped
+            let join_tx = spawn(async move {
+                info!("writing payload");
+                tx.write_all(&payload_cloned).await.unwrap();
+                info!("payload fully written");
+                tx
+            });
+
+            let mut incoming_buffer = vec![0; PAYLOAD_SIZE];
+            let size = rx.read_exact(incoming_buffer.as_mut_slice()).await.unwrap();
+
+            // check that data is correct up to the size
+            assert_eq!(payload.len(), size);
+
+            // using assert!() to avoid MB of data being shown in the logs
+            assert!(payload == incoming_buffer);
+
+            let _ = join_tx.await.unwrap();
+            second_node.context.stop().await?;
+            first_node.context.stop().await?;
+
+            Ok(())
+        };
+
+        timeout(Duration::from_secs(90), test_body)
+            .await
+            .unwrap_or_else(|_| Err(Error::new(Origin::Node, Kind::Timeout, "Test timed out")))
+    });
+
+    result.unwrap();
+
+    // to avoid tokio panic when dropping the runtime
+    // shouldn't be necessary since this should be the default rust behavior
+    drop(runtime_cloned);
+}
+
+#[test]
+fn portal_connection_drop_packets() {
+    // Drop even packets after 32 packets (to allow for the initial
+    // handshake to complete).
+    // This test checks that:
+    //   - connection is interrupted when a failure is detected
+    //   - the portion of the received data matches with the sent data.
+    //
+
+    test_portal_payload_transfer(Disruption::DropPacketsAfter(32));
+}
+
+#[test]
+fn portal_connection_change_packet_order() {
```

Comment:
These two tests a flaky for me locally. Example of a failed run:
```
2024-02-20T00:51:51.562550Z  INFO ockam_node::node: Initializing ockam node
2024-02-20T00:51:51.595575Z  WARN new in-memory node{node_name="exciting-krait"}:create_node_manager{node_name="exciting-krait"}: ockam_api::nodes::service: no policy access control set for resource 'echo' and action: 'handle_message'
2024-02-20T00:51:51.595634Z  INFO new in-memory node{node_name="exciting-krait"}:create_node_manager{node_name="exciting-krait"}: ockam_api::nodes::service: created a node manager for the node: exciting-krait
2024-02-20T00:51:51.595710Z  INFO ockam_node::node: Initializing ockam node
2024-02-20T00:51:51.626249Z  WARN new in-memory node{node_name="fresh-eagle"}:create_node_manager{node_name="fresh-eagle"}: ockam_api::nodes::service: no policy access control set for resource 'echo' and action: 'handle_message'
2024-02-20T00:51:51.626300Z  INFO new in-memory node{node_name="fresh-eagle"}:create_node_manager{node_name="fresh-eagle"}: ockam_api::nodes::service: created a node manager for the node: fresh-eagle
2024-02-20T00:51:51.626386Z  INFO create_outlet{socket_addr=127.0.0.1:59732 worker_addr=0#outlet alias="outlet_alias" reachable_from_default_secure_channel=true access_control=Some(AllowAll) policy_expression=None}: ockam_api::nodes::service::portals: Handling request to create outlet portal at 127.0.0.1:59732 with worker 0#outlet
2024-02-20T00:51:51.626648Z  INFO ockam_api::nodes::service::portals: Handling request to create inlet portal
2024-02-20T00:51:51.626748Z  WARN ockam_api::nodes::service: no policy access control set for resource 'inlet_alias' and action: 'handle_message'
2024-02-20T00:51:51.636427Z  INFO ockam_identity::secure_channel::handshake::handshake_worker: Initialized SecureChannel initiator at local: 0#14fb081a7e77ae3648cd392f0b34b7d1, remote: 0#534c108bbe197095d08752a563a99cea
2024-02-20T00:51:51.636629Z  INFO portals: inlet_status: InletStatus { bind_addr: "127.0.0.1:59736", worker_addr: Some("9ded0fed17848363bbe9f3bc81a8b014"), alias: "inlet_alias", payload: None, outlet_route: Some("0#14fb081a7e77ae3648cd392f0b34b7d1 => 0#outlet"), status: Up, outlet_addr: "/ip4/127.0.0.1/tcp/59735/secure/api/service/outlet" }
2024-02-20T00:51:51.638585Z  INFO ockam_identity::secure_channel::handshake::handshake_worker: Initialized SecureChannel responder at local: 0#3f0bd8deaa865599c0d0adb0354aef5d, remote: 0#52c8955bd79ad30db8b37b38771af25f
2024-02-20T00:51:52.231253Z  INFO ockam_transport_tcp::portal::portal_worker: Creating new "inlet" at internal: 0#dbcd6458736e0b62aff474120b564750, remote: 0#ed109b797230fc5ee3ad99df3183c7d5
2024-02-20T00:51:52.232052Z  INFO ockam_transport_tcp::portal::portal_worker: Creating new "outlet" at internal: 0#613d97413bff974e876420409c419bee, remote: 0#fc07dab9595a4444d9aa38c176df610c
2024-02-20T00:51:52.478652Z  INFO portals::utils: Dropping packet 33 of size 30979
2024-02-20T00:51:52.489597Z  WARN ockam_transport_tcp::portal::portal_worker: Received packet with counter 32 while expecting 31, disconnecting
2024-02-20T00:51:52.492589Z  INFO portals::utils: Dropping packet 35 of size 34051
2024-02-20T00:51:52.497882Z  INFO portals::utils: Dropping packet 37 of size 1280
2024-02-20T00:51:52.513189Z  INFO portals::utils: Dropping packet 39 of size 8448
2024-02-20T00:51:52.528619Z  INFO portals::utils: Dropping packet 41 of size 15616
2024-02-20T00:51:52.687754Z  INFO portals::utils: Dropping packet 33 of size 49411
2024-02-20T00:51:52.718154Z  INFO portals::utils: Dropping packet 35 of size 49411
2024-02-20T00:51:52.748497Z  INFO portals::utils: Dropping packet 37 of size 49411
2024-02-20T00:51:52.778176Z  INFO portals::utils: Dropping packet 39 of size 49411
2024-02-20T00:51:52.819887Z  INFO portals::utils: Dropping packet 41 of size 49411
2024-02-20T00:51:52.833637Z  INFO portals::utils: Dropping packet 43 of size 49411
2024-02-20T00:51:52.859010Z  INFO portals::utils: Dropping packet 45 of size 49411
2024-02-20T00:51:52.884267Z  INFO portals::utils: Dropping packet 47 of size 49411
2024-02-20T00:51:52.910052Z  INFO portals::utils: Dropping packet 49 of size 49411
2024-02-20T00:51:52.936037Z  INFO portals::utils: Dropping packet 51 of size 49411
2024-02-20T00:51:52.961632Z  INFO portals::utils: Dropping packet 53 of size 49411
2024-02-20T00:51:52.986887Z  INFO portals::utils: Dropping packet 55 of size 49411
2024-02-20T00:51:53.011973Z  INFO portals::utils: Dropping packet 57 of size 49411
2024-02-20T00:51:53.037230Z  INFO portals::utils: Dropping packet 59 of size 49411
2024-02-20T00:51:53.062295Z  INFO portals::utils: Dropping packet 61 of size 49411
2024-02-20T00:51:53.087467Z  INFO portals::utils: Dropping packet 63 of size 49411
2024-02-20T00:51:53.112562Z  INFO portals::utils: Dropping packet 65 of size 49411
2024-02-20T00:51:53.138212Z  INFO portals::utils: Dropping packet 67 of size 49411
2024-02-20T00:51:53.163637Z  INFO portals::utils: Dropping packet 69 of size 49411
2024-02-20T00:51:53.515107Z  INFO ockam_transport_tcp::portal::portal_worker: "outlet" at: 0#613d97413bff974e876420409c419bee stopped due to connection drop
2024-02-20T00:51:53.532141Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.547256Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.561370Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.561580Z  INFO portals::utils: Dropping packet 71 of size 49411
2024-02-20T00:51:53.561612Z  INFO portals::utils: Dropping packet 73 of size 49411
2024-02-20T00:51:53.561638Z  INFO portals::utils: Dropping packet 75 of size 49411
2024-02-20T00:51:53.561664Z  INFO portals::utils: Dropping packet 77 of size 49411
2024-02-20T00:51:53.561760Z  INFO portals::utils: Dropping packet 79 of size 49411
2024-02-20T00:51:53.561783Z  INFO portals::utils: Dropping packet 81 of size 49411
2024-02-20T00:51:53.574727Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.587907Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.601008Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.614031Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.615472Z  INFO portals::utils: Dropping packet 83 of size 49411
2024-02-20T00:51:53.615511Z  INFO portals::utils: Dropping packet 85 of size 49411
2024-02-20T00:51:53.615533Z  INFO portals::utils: Dropping packet 87 of size 49411
2024-02-20T00:51:53.615611Z  INFO portals::utils: Dropping packet 89 of size 49411
2024-02-20T00:51:53.615651Z  INFO portals::utils: Dropping packet 91 of size 49411
2024-02-20T00:51:53.616615Z  INFO portals::utils: Dropping packet 93 of size 49411
2024-02-20T00:51:53.627049Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.640025Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.653003Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.665925Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.678895Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.691891Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.692035Z  INFO portals::utils: Dropping packet 95 of size 49411
2024-02-20T00:51:53.692057Z  INFO portals::utils: Dropping packet 97 of size 49411
2024-02-20T00:51:53.692073Z  INFO portals::utils: Dropping packet 99 of size 49411
2024-02-20T00:51:53.701925Z  INFO portals::utils: Dropping packet 101 of size 49411
2024-02-20T00:51:53.704906Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.718020Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.727856Z  INFO portals::utils: Dropping packet 103 of size 49411
2024-02-20T00:51:53.730906Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.743816Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.753337Z  INFO portals::utils: Dropping packet 105 of size 49411
2024-02-20T00:51:53.757068Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.770130Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.778894Z  INFO portals::utils: Dropping packet 107 of size 49411
2024-02-20T00:51:53.783121Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.795962Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.804317Z  INFO portals::utils: Dropping packet 109 of size 49411
2024-02-20T00:51:53.809243Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.822398Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.829949Z  INFO portals::utils: Dropping packet 111 of size 49411
2024-02-20T00:51:53.835567Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.848711Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.855675Z  INFO portals::utils: Dropping packet 113 of size 49411
2024-02-20T00:51:53.861760Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.874731Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.881604Z  INFO portals::utils: Dropping packet 115 of size 49411
2024-02-20T00:51:53.887734Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.900670Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.906824Z  INFO portals::utils: Dropping packet 117 of size 49411
2024-02-20T00:51:53.913707Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.926755Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.932773Z  INFO portals::utils: Dropping packet 119 of size 49411
2024-02-20T00:51:53.939619Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.952828Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.958092Z  INFO portals::utils: Dropping packet 121 of size 49411
2024-02-20T00:51:53.966032Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.979297Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:53.983432Z  INFO portals::utils: Dropping packet 123 of size 49411
2024-02-20T00:51:53.992302Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.005457Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.009161Z  INFO portals::utils: Dropping packet 125 of size 49411
2024-02-20T00:51:54.018403Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.031519Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.034394Z  INFO portals::utils: Dropping packet 127 of size 49411
2024-02-20T00:51:54.044423Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.057463Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.060190Z  INFO portals::utils: Dropping packet 129 of size 49411
2024-02-20T00:51:54.070572Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.083145Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.085852Z  INFO portals::utils: Dropping packet 131 of size 49411
2024-02-20T00:51:54.096296Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.109164Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.122181Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.122821Z  INFO portals::utils: Dropping packet 133 of size 49411
2024-02-20T00:51:54.135312Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.136863Z  INFO portals::utils: Dropping packet 135 of size 49411
2024-02-20T00:51:54.147869Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.160807Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.162383Z  INFO portals::utils: Dropping packet 137 of size 49411
2024-02-20T00:51:54.173878Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.186686Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.187682Z  INFO portals::utils: Dropping packet 139 of size 49411
2024-02-20T00:51:54.199544Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.212504Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.213377Z  INFO portals::utils: Dropping packet 141 of size 49411
2024-02-20T00:51:54.225237Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.238435Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.238843Z  INFO portals::utils: Dropping packet 143 of size 49411
2024-02-20T00:51:54.251481Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.264485Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.264652Z  INFO portals::utils: Dropping packet 145 of size 49411
2024-02-20T00:51:54.277477Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.290445Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.290514Z  INFO portals::utils: Dropping packet 147 of size 49411
2024-02-20T00:51:54.303398Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.316195Z  INFO portals::utils: Dropping packet 149 of size 49411
2024-02-20T00:51:54.317614Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.341369Z  INFO portals::utils: Dropping packet 151 of size 49411
2024-02-20T00:51:54.343408Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.367143Z  INFO portals::utils: Dropping packet 153 of size 49411
2024-02-20T00:51:54.368275Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.392522Z  INFO portals::utils: Dropping packet 155 of size 49411
2024-02-20T00:51:54.394317Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.418323Z  INFO portals::utils: Dropping packet 157 of size 49411
2024-02-20T00:51:54.419687Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.443989Z  INFO portals::utils: Dropping packet 159 of size 49411
2024-02-20T00:51:54.445455Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.469615Z  INFO portals::utils: Dropping packet 161 of size 49411
2024-02-20T00:51:54.471157Z  WARN ockam_identity::secure_channel::decryptor: operation failed for address 0#fc07dab9595a4444d9aa38c176df610c (origin: Node, kind: NotFound, source location: /Users/sanjo/sources/ockam/implementations/rust/ockam/ockam_node/src/router/utils.rs:26:19) forwarding decrypted message from 0#3f0bd8deaa865599c0d0adb0354aef5d
2024-02-20T00:51:54.493009Z  INFO ockam_transport_tcp::portal::portal_worker: "inlet" at: 0#dbcd6458736e0b62aff474120b564750 stopped due to connection drop
2024-02-20T00:51:54.493035Z ERROR ockam_node::relay::worker_relay: Error encountered during '0#dbcd6458736e0b62aff474120b564750' message handling: failed to receive a malformed message (origin: Transport, kind: Serialization, source location: implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_worker.rs:443:44)
2024-02-20T00:51:54.493210Z ERROR ockam_node::node: message1="A fatal error occurred: panicked at implementations/rust/ockam/ockam_api/tests/portals.rs:605:67:\ncalled `Result::unwrap()` on an `Err` value: Os { code: 54, kind: ConnectionReset, message: \"Connection reset by peer\" }."
2024-02-20T00:51:54.493230Z ERROR ockam_node::node: message2="Please report this issue, with a copy of your logs, to https://github.com/build-trust/ockam/issues."
```

Reply:
@SanjoDeundiak 

I can't reproduce it, do you remember when did it fail?

Reply:
managed to reproduce it on a mac :+1: